src/mlpack/tests/init_rules_test.cpp

```diff
@@ -285,4 +286,47 @@ BOOST_AUTO_TEST_CASE(NetworkInitTest)
   BOOST_REQUIRE_EQUAL(gaussianModel.Parameters().n_elem, 42);
 }
 
+/**
+ * Simple test of the GlorotInitialization class for uniform distribution
+ */
+BOOST_AUTO_TEST_CASE(GlorotInitUniformTest)
+{
+  arma::mat weights;
+  arma::cube weights3d;
+
+  GlorotInitialization<true> glorotInit;
+
+  glorotInit.Initialize(weights, 100, 100);
+  glorotInit.Initialize(weights3d, 100, 100, 2);
+
+  BOOST_REQUIRE_EQUAL(weights.n_rows, 100);
+  BOOST_REQUIRE_EQUAL(weights.n_cols, 100);
+
+  BOOST_REQUIRE_EQUAL(weights3d.n_rows, 100);
+  BOOST_REQUIRE_EQUAL(weights3d.n_cols, 100);
+  BOOST_REQUIRE_EQUAL(weights3d.n_slices, 2);
+}
+
+/**
+ * Simple test of the GlorotInitialization class for normal distribution
+ */
+BOOST_AUTO_TEST_CASE(GlorotInitNormalTest)
+{
+  arma::mat weights;
+  arma::cube weights3d;
+
+  GlorotInitialization<false> glorotInit;
```

Comment:
Do you think we could define ``using XavierInitialization = GlorotInitialization<false>`` in the header file. Another idea is to call the class ``GlorotInitializationType`` and define ``XavierInitialization`` and ``GlorotInitialization``, that way we can use the method without the template specifier.

Reply:
Sorry, but I couldn't understand the last part. Maybe, I'm too naive but still it'll be helpful if you can elaborate on the part of `GlorotInitializationType`

Reply:
In this case we use ``using`` to create an alias, so that we can use ``XavierInitialization`` and ``GlorotInitialization`` without the ``<..>``, where ``XavierInitialization`` is ``GlorotInitializationType<false>`` and ``GlorotInitialization`` is ``GlorotInitializationType<true>``, note that I used ``GlorotInitializationType`` and not ``GlorotInitialization`` to avoid a redefinition issue here.

We can put that at the end of the class defenition, here is an example: https://github.com/mlpack/mlpack/blob/486be65dab54e8f0fba4066827c8358ffdea3d4e/src/mlpack/methods/ann/layer/base_layer.hpp#L125-L137

Reply:
Thanks, it means I was understanding it correctly.